projects/components/src/table/table.component.ts

```diff
@@ -330,6 +343,85 @@ export class TableComponent
     this.rowStateSubject.complete();
     this.columnStateSubject.complete();
     this.columnConfigsSubject.complete();
+
+    this.mouseMoveFunction && this.mouseMoveFunction();
+    this.mouseUpFunction && this.mouseUpFunction();
+  }
+
+  public trackItem(_index: number, column: TableColumnConfigExtended): string {
+    return `${column.id}-${column.width}`;
+  }
+
+  public queryHeaderRowElement(): HTMLElement {
+    return this.elementRef.nativeElement.querySelector('cdk-header-row');
+  }
+
+  public queryHeaderCellElement(index: number): HTMLElement {
```

Comment:
Why are these public? Also, could we get this through angular? For example, by iterating through a view children query.

Reply:
I tried using ViewChildren, but it doesn't update correctly. Even subscribing to `changes` on the QueryList gave wrong results because of the ngFor with dynamic content, I assume. It would start with the correct columns, then as we modify them they get removed from the QueryList and never return.

Reply:
Hmm strange. Anyway, can we make this private? Looks to only be called from a private method.

Reply:
The only reason I made it public was so I could override the method during test. I don't like modifying code just for a test, but the alternative is importing the CDK into the test and building the full DOM there, which is a bigger undertaking.

Reply:
Agree that alternative is worse. Another potential alternative which we've used elsewhere for similar issues is `DomElementMeasurerService` - allowing mocking measurements without mocking the dom itself.